lib/puppet/node.rb

```diff
@@ -108,12 +108,29 @@ def fact_merge
   # Merge any random parameters into our parameter list.
   def merge(params)
     params.each do |name, value|
-      @parameters[name] = value unless @parameters.include?(name)
+      if @parameters.include?(name)
+        Puppet::Util::Warnings.warnonce("The node paramters #{name} for node #{@parameters["hostname"]} was already set to #{@parameters[name]}. It could not be set to #{value}")
```

Comment:
Are we changing the behavior of this? Because if I'm understanding the original code correctly then the parameter would only be updated if it was not already there:

`@parameters[name] = value unless @parameters.include?(name)`